### PR TITLE
don't dropout in eval mode

### PR DESCRIPTION
**Pull request by YassineYousfi**
To give the model more chances to figure out if it's being trained or deployed :) (c.f. https://twitter.com/karpathy/status/1635049541534879745)

### CODE REVIEW

**Comment by karpathy** [comment]
Whoa hold on this is an actual bug in my implemtnation right? Basically we are applying dropout at evaluation time?

**Comment by YassineYousfi** [comment]
Yes. I only noticed it recently while training some models with dropout.

**Comment by karpathy** [comment]
😱 😱 😱
:'(
... shook. calling functional directly considered dangerous
ty

**Comment by YassineYousfi** [comment]
You mean after this PR was merged?
There should not be any difference in the first call of estimate_loss() after this PR was merged.

**Comment by prshnthrv** [comment]
I think before this PR. This PR only changes if we are using the flash attention module. In my setup, I am not using flash attention. So this fix should not have any impact on my evaluation.

**Comment by prshnthrv** [comment]
If you run
$ python train.py config/eval_gpt2.py
but change the dropout value in https://github.com/karpathy/nanoGPT/blob/master/train.py#L55 to 0.1 or 0.2, the val_loss is different as I mentioned earlier.

**Comment by YassineYousfi** [comment]
Yes I just did that on current master (with the shakespeare dataset, with and without flash attention) and the losses were the same.

**Comment by prshnthrv** [comment]
Interesting. I’m using openwebtext. I can try with Shakespeare as well. But was trying to match the owt numbers specified.
